athenahealth/drivers_license.go

```diff
@@ -50,3 +52,47 @@ func (h *HTTPClient) AddPatientDriversLicenseDocument(ctx context.Context, patie
 	}, nil
 
 }
+
+type AddPatientDriversLicenseDocumentReaderOptions struct {
+	DepartmentID string
+	Image        io.Reader
+}
+
+type addPatientDriversLicenseDocumentReaderResponse struct {
+	Success bool `json:"success"`
+}
+
+type AddPatientDriversLicenseDocumentReaderResult struct {
+	Success bool
+}
+
+// AddPatientDriversLicenseDocumentReader - performs the same operation as AddPatientDriversLicenseDocument except is more memory efficient
+// by streaming the image into the request, assuming you haven't already read the
+// entire image into memory
+// POST /v1/{practiceid}/patients/{patientid}/driverslicense
+// https://docs.athenahealth.com/api/api-ref/drivers-license#Add-patient's-driver's-license-document
+func (h *HTTPClient) AddPatientDriversLicenseDocumentReader(ctx context.Context, patientID string, opts *AddPatientDriversLicenseDocumentReaderOptions) (*AddPatientDriversLicenseDocumentReaderResult, error) {
+	if opts == nil {
+		panic("opts is nil")
```

Comment:
unnecessary panic, could return error

Reply:
Yea that's fair - there are lots of panic calls in this repo for when nil is unexpectedly passed. I think we should track this as an [issue](https://github.com/eleanorhealth/go-athenahealth/issues) to convert all panics to errors.